### PR TITLE
Boilerplate config changes 2

### PR DESCRIPTION
**Pull request by supertick**

### CODE REVIEW

File: doc/service-life-cycle.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    [*] --> create
+    create --> load
+
+```
+
+### start(name, type)
+Creates and starts a service with the given name and type
+
+### load
+Starts loading the heirarchy of configuration
```
Comment (AutonomicPerfectionist) [inline doc/service-life-cycle.md]: heirarchy -> hierarchy

File: src/main/java/org/myrobotlab/document/connector/AbstractConnector.java [hunk 2]
```diff
@@ -17,7 +17,7 @@
  * service.
  * 
  */
-public abstract class AbstractConnector extends Service implements DocumentPublisher, DocumentConnector {
+public abstract class AbstractConnector extends Service<ServiceConfig> implements DocumentPublisher, DocumentConnector {
```
Comment (AutonomicPerfectionist) [inline src/main/java/org/myrobotlab/document/connector/AbstractConnector.java:20]: I think all abstract classes should have a type variable so downstream classes can change the config type, even if none do right now. So it should be AbstractConnector<C extends ServiceConfig> extends Service<C>, and all subclasses should provide the config type when extending it

File: src/main/java/org/myrobotlab/framework/Service.java [hunk 3]
```diff
@@ -1445,49 +1478,30 @@ public ServiceConfig getConfig() {
...
           // so doesn't conflict with remote routes
-          Listener newConfigListener = new Listener(
-                  listener.topicMethod,
-                  CodecUtils.removeEnd(
-                          listener.callbackName,
-                          '@' + Platform.getLocalInstance().getId()
-                  ),
-                  listener.callbackMethod
-          );
+          Listener newConfigListener = new Listener(listener.topicMethod, CodecUtils.removeEnd(listener.callbackName, '@' + Platform.getLocalInstance().getId()),
```
Comment (AutonomicPerfectionist) [inline src/main/java/org/myrobotlab/framework/Service.java]: Just a note, we should probably move the removeEnd method to StringUtils, being it doesn't have anything to do with codecs

File: src/main/java/org/myrobotlab/framework/Service.java [hunk 4]
```diff
@@ -148,7 +155,6 @@ public abstract class Service implements Runnable, Serializable, ServiceInterfac
   /**
    * the last config applied to this service
    */
```
Comment (AutonomicPerfectionist) [inline src/main/java/org/myrobotlab/framework/Service.java]: Remove the dangling Java doc

File: doc/service-life-cycle.md (hunk 1 above)
Comment (supertick) [reply doc/service-life-cycle.md]: :+1:

Comment (supertick) [reply doc/service-life-cycle.md]: :+1:

File: src/main/java/org/myrobotlab/document/connector/AbstractConnector.java (hunk 2 above)
Comment (supertick) [reply src/main/java/org/myrobotlab/document/connector/AbstractConnector.java:20]: :+1:

File: src/main/java/org/myrobotlab/framework/Service.java (hunk 3 above)
Comment (supertick) [reply src/main/java/org/myrobotlab/framework/Service.java]: :+1:

File: src/main/java/org/myrobotlab/framework/Service.java (hunk 4 above)
Comment (supertick) [reply src/main/java/org/myrobotlab/framework/Service.java]: :+1:

**Pull request merged**